numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -379,11 +379,16 @@ array_implement_c_array_function_creation(
         return Py_NotImplemented;
     }
 
-    PyObject *relevant_args = PyTuple_Pack(1,
-            PyDict_GetItem(kwargs, npy_ma_str_like));
-    if (relevant_args == NULL) {
+    PyObject *like_arg = PyDict_GetItem(kwargs, npy_ma_str_like);
+    if (like_arg == NULL) {
         return NULL;
     }
+    else if (!PyObject_HasAttrString(like_arg, "__array_function__")) {
```

Comment:
This will lead to weird situations where this check passes, but `__array_function__` is not actually present - to behave accurately, this should use `get_array_function(like_arg)`

Reply:
Good idea, I'll change that.

Reply:
This change is in too.